### PR TITLE
restricted number of beats allowed based on the current time signature

### PR DESCRIPTION
**Pull request by Sandermoen**

### CODE REVIEW

File: src/pages/index.js [hunk 1]
```diff
@@ -35,13 +35,39 @@ const Home = ({ originalLayout, error }) => {
     return clearInterval(useInterval);
   }, []);
 
+  useEffect(() => {
```
Comment (rgavinc) [inline src/pages/index.js]: Great use of `useEffect`!

File: src/pages/index.js [hunk 2]
```diff
@@ -35,13 +35,39 @@ const Home = ({ originalLayout, error }) => {
...
+      layout.forEach(row => {
+        beatLength < beatsMeasure ? row.beats.push(false) : row.beats.pop();
+      });
+      beatLength = layout[0].beats.length;
+    }
+    return setLayout(layout);
+  };
+
+  const updateBeats = (layout, remove) => {
+    layout.forEach(row => {
```
Comment (rgavinc) [inline src/pages/index.js]: - [ ] Let's use the [filter array function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) instead here.

File: src/pages/index.js [hunk 3]
```diff
@@ -35,13 +35,39 @@ const Home = ({ originalLayout, error }) => {
...
     if (!active) {
       setCount(1);
     }
     setPlay(active);
   };
 
+  const validateBeats = layout => {
+    let beatLength = layout[0].beats.length;
+    while (!Number.isInteger(beatLength / beatsMeasure)) {
+      layout.forEach(row => {
```
Comment (rgavinc) [inline src/pages/index.js]: - [ ] Let's use the [filter array function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) instead here.

File: src/pages/index.js (hunk 1 above)
Comment (Sandermoen) [reply src/pages/index.js]: Thanks :)

**Pull request merged**